src/ui/pages/AssetInfo/AssetChart/AssetChart.tsx

```diff
@@ -0,0 +1,300 @@
+      ? Math.max(...chartPoints.map(([, value]) => value))
+      : null;
+  }, [chartPoints]);
+
+  const minChartPointValue = useMemo(() => {
+    return chartPoints.length
+      ? Math.min(...chartPoints.map(([, value]) => value))
+      : null;
+  }, [chartPoints]);
+
+  useEffect(() => {
+    const ctx = canvasRef.current?.getContext('2d');
+    if (!ctx) {
+      return;
+    }
+
+    chartRef.current = new Chart(ctx, {
+      ...DEFAULT_CONFIG,
+      data: {
+        datasets: [
+          {
+            data: [],
+            hidden: true,
+            segment: {
+              borderColor: (ctx) =>
+                getSegmentColor({
+                  ctx,
+                  startRangeIndex: startRangeIndexRef.current,
+                  endRangeIndex: endRangeIndexRef.current,
+                  chartPoints: chartPointsRef.current,
+                  theme: themeRef.current,
+                }),
+            },
+          },
+        ],
+      },
+      options: {
+        ...DEFAULT_CONFIG.options,
+        onHover: (_, __, chart) => {
+          chart.update();
+        },
+      },
+      plugins: [
+        {
+          id: 'mouseSelectPlugin',
+          beforeEvent: (chart, args) => {
+            if (args.event.type === 'mousedown') {
+              startRangeIndexRef.current =
+                chart.getActiveElements()[0]?.index ?? null;
+              startRangeXRef.current = args.event.x;
+            }
+            if (args.event.type === 'mousemove') {
+              endRangeIndexRef.current =
+                chart.getActiveElements()[0]?.index ?? null;
+            }
+            if (
+              args.event.type === 'mouseout' ||
+              args.event.type === 'mouseup'
+            ) {
+              endRangeIndexRef.current = null;
+              startRangeIndexRef.current = null;
+              startRangeXRef.current = null;
+            }
+            onRangeSelectEvent(
+              getSortedRangeIndexes({
+                startRangeIndex: startRangeIndexRef.current,
+                endRangeIndex: endRangeIndexRef.current,
+              })
+            );
+          },
+          afterEvent: (chart, args) => {
+            if (args.event.type === 'mouseout') {
+              chart.update();
+            }
+          },
+        },
+        drawDotPlugin({
+          getStartRangeIndex: () => startRangeIndexRef.current,
+          getTheme: () => themeRef.current,
+        }),
+        drawRangePlugin({
+          getStartRangeX: () => startRangeXRef.current,
+          getTheme: () => themeRef.current,
+        }),
+      ],
+    });
+
+    return () => {
+      chartRef.current?.destroy();
+    };
+  }, [onRangeSelectEvent]);
+
+  if (!equal(chartPointsRef.current, chartPoints) && chartRef.current) {
+    updateChartPoints({
+      chart: chartRef.current,
+      prevPoints: chartPointsRef.current,
+      nextPoints: chartPoints,
+      theme,
+    });
+    chartPointsRef.current = [...chartPoints];
```

Comment:
```suggestion
    chartPointsRef.current = chartPoints;
```
? Or am I missing the point? %)